### PR TITLE
copy and build vignettes in docs/articles

### LINKED ISSUE #172: vignette images not loading
This is so useful, I can't believe I've only just seen it!

One hiccup; the images in our vignettes don't load, e.g.: http://goldingn.github.io/zoon/articles/Building_a_module.html

That's because the file paths in the html generated by `pkgdown::build_site()` are absolute: https://github.com/goldingn/zoon/blob/gh-pages/reference/articles/Building_a_module.html#L187

I'm guessing this is because `build_articles()` calls `rmarkdown::html_document()` with the argument `self_contained = FALSE`: https://github.com/hadley/pkgdown/blob/master/R/build-articles.R#L96, although it's not clear to me why they get the full filepath  rather than relative to wherever `build_site()` is executed.

### PR DESCRIPTION
**Pull request by jayhesselberth**
fixes #172 

Strangely PNGs generated from the render all have the format `README-chunk_label-number.png` (e.g., `README-plot-1.png`), despite the fact that many of the images don't come from a README per se.

I couldn't find where the `README` part is coming from; I presume this is a default on the `rmarkdown` side.

I think a better filename would be `pkg$vignettes$name-chunk_label-number.png`.

### CODE REVIEW

**Comment by jayhesselberth** [comment]
talk about pre-coffee ... was setting `fig.path = README-` in `knitr::opts_chunk`.

**Comment by hadley** [comment]
I think you might need to copy all files? What happens if (e.g.) you're linking to an image that you've saved in the vignettes directory?

File: R/util.r [hunk 1]
```diff
@@ -109,3 +109,18 @@ print_yaml <- function(x) {
...
   cat(yaml::as.yaml(x), "\n", sep = "")
 }
+
+copy_dir <- function(from, to) {
+
+  from_dirs <- list.dirs(from, full.names = FALSE, recursive = TRUE)
+  from_dirs <- from_dirs[from_dirs != '']
+
+  to_dirs <- file.path(to, from_dirs)
+  sapply(to_dirs, dir.create)
```
Comment (hadley) [inline R/util.r]: Can you please use `purrr::walk` here?

**Comment by jayhesselberth** [comment]
If a package `README.Rmd` creates PNGs that are stored at the top level of the package, I think those need to copied to `docs/articles` as well. Does that sound right?

**Comment by jayhesselberth** [comment]
And if those PNGs need to be copied, how do suggest dealing robustly with copying them over? What if they're not stored at the top level, but in an `img/` directory? Parse the README.Rmd to determine where PNGs would be?

**Comment by jayhesselberth** [comment]
ok so looks like top level PNGs *should* be rendered by `build-home.R` (so no copying necessary), but aren't currently. `build-home.R` currently only checks for `README.md` and `index.md`, despite the fact the current docs say you can also use `README.Rmd` and `index.Rmd`. Anyway I'll leave this a separate issue.

**Comment by hadley** [comment]
I'm working on the home Rmd stuff now

**Pull request merged**